### PR TITLE
(SIMP-3409) mcstransd service problem when selinux is disabled

### PR DESCRIPTION
**Pull request by lnemsick-simp**
Ensure mcstrans(d) and restorecond services are not running
when selinux is not enabled.

This uses modern, selinux facts and was acceptance tested with both
puppet 4.7.0 and puppet 4.8.2.

SIMP-3409 #close

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -12,15 +12,44 @@
...
+# @param package_ensure The ensure status of packages to be installed
+#
 class selinux (
   Selinux::State         $ensure               = simplib::lookup('simp_options::selinux', { 'default_value' => true }),
   Boolean                $manage_utils_package = true,
   String                 $package_ensure       = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
-  Enum['targeted','mls'] $mode                 = 'targeted'
+  Enum['targeted','mls'] $mode                 = 'targeted',
+  # defaults are in module data
+  Boolean $manage_mcstrans_package,
```
Comment (trevor-vaughan) [inline manifests/init.pp]: These will need to be pushed to the top of the stack so that the linter doesn't complain.

**Comment by kendall-moore** [comment]
Just as a consideration:

Best practices in component module design would be to abstract out the package logic into a separate manifest, perhaps something like selinux::install and the same for the service logic into something like selinux::service.   These would be declared private and would be contained within selinux::init.

While that's not necessary and wouldn't functionally change anything, it could help in keeping consistent across our component module baseline and also keep us close in line with Puppet's published best practices.

As a quick reference, here are Puppet's best practices suggestions for module design: [https://docs.puppet.com/puppet/5.0/bgtm.html#structuring-your-module](https://docs.puppet.com/puppet/5.0/bgtm.html#structuring-your-module)

File: manifests/service.pp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    $_aux_service_ensure = 'running'
+  }
+
+  if $::selinux::manage_mcstrans_service {
+    service { $::selinux::mcstrans_service_name:
+      ensure     => $_aux_service_ensure,
+      enable     => true,
+      hasrestart => true,
+      hasstatus  => false,
+      require    => Package[$::selinux::mcstrans_package_name]
```
Comment (kendall-moore) [inline manifests/service.pp]: This should require the class, not the resource.

require => Class['selinux::install']

File: manifests/service.pp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  }
+
+  if $::selinux::manage_restorecond_service {
+    service { 'restorecond':
+      ensure     => $_aux_service_ensure,
+      enable     => true,
+      hasrestart => true,
+      hasstatus  => false,
+      require    => Package[$::selinux::restorecond_package_name]
```
Comment (kendall-moore) [inline manifests/service.pp]: This should require the class, not the resource.

require => Class['selinux::install']

File: manifests/init.pp [hunk 4]
```diff
@@ -12,39 +12,55 @@
...
-  }
-
-  $utils_packages = [
-    'checkpolicy',
-    'policycoreutils-python',
-  ]
-  if $manage_utils_package {
-    ensure_resource('package', $utils_packages, { 'ensure' => $package_ensure })
-  }
+  include 'selinux::install'
```
Comment (kendall-moore) [inline manifests/init.pp]: Should be contain instead of include:

contain 'selinux::install'

File: manifests/init.pp [hunk 5]
```diff
@@ -12,39 +12,55 @@
...
+  include 'selinux::config'
```
Comment (kendall-moore) [inline manifests/init.pp]: Should be contain instead of include:

contain 'selinux::config'

File: manifests/init.pp [hunk 6]
```diff
@@ -12,39 +12,55 @@
...
+  include 'selinux::service'
```
Comment (kendall-moore) [inline manifests/init.pp]: Should be contain instead of include:

contain 'selinux::service'

**Pull request merged**